### PR TITLE
add unit, regression, edge case testing

### PR DESCRIPTION
**Pull request by gibsongreen**
Added 63 new unit tests across 4 test modules to improve coverage of core slitlessutils functionality:

**test_wfss_simulated.py**
- Validates simulated WFSS data creation and orientation handling
- Tests detector configuration keywords (NAXIS, CRPIX, CRVAL, spectral orders)
- Verifies WFSSCollection CSV loading and iteration
- Edge case testing for coordinate systems and orientation angles

**test_sources.py**
- Tests SourceCollection loading from segmentation maps
- Validates source properties (segid, pixels, WCS)
- Tests filtering parameters (minpix, maglim)
- Verifies spectral parameter propagation
- Edge cases: sources at image edges, single-pixel sources, high segid values

**test_photometry.py**
- Tests Throughput loading from telescope/instrument/filter keys
- Validates SED creation, data access, and normalization
- Tests photometric workflows and integration
- Edge cases: sharp filter edges, wavelength gaps, low transmission values

**test_config.py**
- Validates Config singleton pattern
- Tests required parameters (fluxscale, fluxunits, compression settings)
- Verifies reference file paths and directory structure
- Tests file suffix definitions

### CODE REVIEW

File: slitlessutils/tests/test_photometry.py [hunk 1]
```diff
@@ -215,4 +215,4 @@ def test_sed_unity_flux_normalization():
     flux = np.ones_like(wave)
     sed = su.photometry.SED(wave, flux)
     sed.normalize(10000., 5.0)
-    assert np.isclose(sed['flam'][50], 5.0, rtol=0.1)
+    assert np.allclose(sed['flam'][50], 5.0, rtol=0.1)
```
Comment (rosteen) [inline slitlessutils/tests/test_photometry.py]: Sorry, I didn't mean to imply that `isclose` needed to change to `allclose`, I'm just in the habit of using `allclose` since I'm usually working with arrays. Probably don't need to change it back though 🤷

**Review by rosteen — APPROVED** [review]
